react/src/components/ImageList.tsx

```diff
@@ -319,53 +325,39 @@ const ImageList: React.FC<{ style?: React.CSSProperties }> = ({ style }) => {
     },
   ];
 
-  const processedImages = useMemo(() => {
-    const sortedImages: EnvironmentImage[] = images
-      ? [...images]
-          .filter(
-            (image): image is EnvironmentImage =>
-              image !== null && image !== undefined,
-          )
-          .sort(
-            (a, b) =>
-              sortBasedOnInstalled(a, b) ||
-              localeCompare(a.humanized_name, b.humanized_name),
-          )
-      : [];
-    return {
-      sortedImages,
-      imageFilterValues: sortedImages.map((image) => {
-        return {
-          namespace: getNamespace(getImageFullName(image) || ''),
-          lang: image.name ? getLang(image.name) : '',
-          baseversion: getBaseVersion(getImageFullName(image) || ''),
-          baseimage:
-            image.tag && image.name ? getBaseImages(image.tag, image.name) : [],
-          constraints:
-            image.tag && image.labels
-              ? getConstraints(
-                  image.tag,
-                  image.labels as { key: string; value: string }[],
-                )
-              : [],
-          isCustomized: image.tag
-            ? image.tag.indexOf('customized') !== -1
-            : false,
-        };
-      }),
-    };
+  const imageFilterValues = useMemo(() => {
+    return defaultSortedImages?.map((image) => {
+      return {
+        installed: image?.installed ? t('environment.Installed') : '',
+        namespace: getNamespace(getImageFullName(image) || ''),
+        lang: image?.name ? getLang(image.name) : '',
+        baseversion: getBaseVersion(getImageFullName(image) || ''),
+        baseimage:
+          image?.tag && image?.name ? getBaseImages(image.tag, image.name) : [],
+        constraints:
+          image?.tag && image?.labels
+            ? getConstraints(
+                image.tag,
+                image.labels as { key: string; value: string }[],
+              )
+            : [],
+        isCustomized: image?.tag
+          ? image.tag.indexOf('customized') !== -1
+          : false,
+      };
+    });
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [images]);
+  }, [defaultSortedImages]);
 
   const filteredImageData = useMemo(() => {
-    const { sortedImages, imageFilterValues } = processedImages;
-    if (_.isEmpty(imageSearch)) return sortedImages;
+    if (_.isEmpty(imageSearch) || imageFilterValues === undefined)
```

Comment:
Is there a case that ImageFilterValues is undefined?

Reply:
Yes, you are right. But we need to handle some `undefined` cases. Please check my latest commit.